api_tests/users/views/test_user_addons.py

```diff
@@ -0,0 +1,422 @@
+
+    def test_account_detail_raises_error_if_not_found(self):
+        wrong_type = self.should_expect_errors()
+        try:
+            if self.user.external_accounts:
+                self.user.external_accounts.pop()
+            self.user.delete_addon(self.short_name, auth=self.auth)
+        except ValueError:
+            # If addon was mandatory -- OSFStorage
+            pass
+        res = self.app.get(
+            self.account_detail_url,
+            auth=self.user.auth,
+            expect_errors=True)
+
+        assert_equal(res.status_code, 404)
+        if not wrong_type:
+            assert_in('Requested addon not enabled', res.json['errors'][0]['detail'])
+        if wrong_type:
+            assert_in('Requested addon unavailable', res.json['errors'][0]['detail'])
+
+    def test_account_detail_raises_error_if_PUT(self):
+        res = self.app.put_json_api(self.account_detail_url, {
+            'id': self.short_name,
+            'type': 'user-external_account-detail'
+            }, auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_account_detail_raises_error_if_PATCH(self):
+        res = self.app.patch_json_api(self.account_detail_url, {
+            'id': self.short_name,
+            'type': 'user-external_account-detail'
+            }, auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_account_detail_raises_error_if_DELETE(self):
+        res = self.app.delete(
+            self.account_detail_url,
+            auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_account_detail_raises_error_if_nonauthenticated(self):
+        res = self.app.get(
+            self.account_detail_url,
+            expect_errors=True)
+
+        assert_equal(res.status_code, 401)
+
+    def test_account_detail_user_cannot_view_other_user(self):
+        other_user = AuthUserFactory()
+        res = self.app.get(
+            self.account_detail_url,
+            auth=other_user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+class UserAddonTestSuiteMixin(UserAddonListMixin, UserAddonDetailMixin, UserAddonAccountListMixin, UserAddonAccountDetailMixin):
+    def set_urls(self):
+        self.set_setting_list_url()
+        self.set_setting_detail_url()
+        self.set_account_list_url()
+        self.set_account_detail_url()
+
+    def should_expect_errors(self, success_types=['OAUTH']):
+        return self.addon_type not in success_types
+
+class UserOAuthAddonTestSuiteMixin(UserAddonTestSuiteMixin):
+    addon_type = 'OAUTH'
+
+    @abc.abstractproperty
+    def AccountFactory(self):
+        pass
+
+
+class UserUnmanageableAddonTestSuiteMixin(UserAddonTestSuiteMixin):
+    addon_type = 'UNMANAGEABLE'
+
+# UNMANAGEABLE
+
+class TestUserForwardAddon(UserUnmanageableAddonTestSuiteMixin, ApiAddonTestCase):
```

Comment:
To test that any addon specified by an incoming behaves as expected; necessary. This is less burdensome than it would be if these were repeated for every addon -- it's the same rationale that went into creating generic OSF tests for addons (see `/website/addons/base/testing/`). 

After resolution of the follow-up ticket, one of these types will go away, as the behavior of `CONFIGURABLE` and `OAUTH` will be equivalent. `UNMANAGEABLE` are addons that you can't interact with via APIv2, and `INVALID` are ones that don't exist.

I realize now that `MANAGEABLE` isn't actually used; two of these types will go away.